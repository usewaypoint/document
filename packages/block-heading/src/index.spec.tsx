import React from 'react';

import { render } from '@testing-library/react';

import { Heading } from '.';

describe('Heading', () => {
  it('renders with default values', () => {
    expect(render(<Heading />).asFragment()).toMatchSnapshot();
  });

  it('renders with style', () => {
    const style = {
      backgroundColor: '#444333',
      color: '#101010',
      fontFamily: 'HEAVY_SANS' as const,
      fontWeight: 'normal' as const,
      padding: {
        top: 15,
        bottom: 10,
        left: 24,
        right: 8,
      },
      textAlign: 'center' as const,
    };
    const props = {
      text: 'Hello world!',
      level: 'h1' as const,
    };
    expect(render(<Heading style={style} props={props} />).asFragment()).toMatchSnapshot();
  });
});
