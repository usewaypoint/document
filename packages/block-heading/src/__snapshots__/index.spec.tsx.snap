// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders with default values 1`] = `
<DocumentFragment>
  <h2
    style="font-weight: bold; margin: 0px; font-size: 24px;"
  />
</DocumentFragment>
`;

exports[`Heading renders with style 1`] = `
<DocumentFragment>
  <h1
    style="color: rgb(16, 16, 16); background-color: rgb(68, 67, 51); font-weight: normal; text-align: center; margin: 0px; font-family: Bahnschrift, "DIN Alternate", "Franklin Gothic Medium", "Nimbus Sans Narrow", sans-serif-condensed, sans-serif; font-size: 32px; padding: 15px 8px 10px 24px;"
  >
    Hello world!
  </h1>
</DocumentFragment>
`;
